CharlieBackend.Core/Models/ResultModel/Result.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+
+namespace CharlieBackend.Core.Models.ResultModel
+{
+    /// <summary>
+    /// Class is used to transfer data or errors information. Methods has to return Result<T> data type
+    /// </summary>
+    /// <typeparam name="T">type of data transferred thrue Result class</typeparam>
+    public class Result<T>
+    {
+        public T TransferredData { get; set; }
+
+        public ErrorData ErrorData { get; set; }
```

Comment:
Rename this to Error

Reply:
ErrorData name is choosen due to Error method name conflict